Henrikm/buildandutestupdates

henrik-me opened the pull request:
minor doc bug fix and adding a few simple scripts to build and restore. also added the beginning of a pre-req validation script. Made an update to the target android version... I believe it's ok, but it may have side effects and should potentially be 26 and not 27 (as that is targeting v.8 which is also updated in this change). Made a few updates to address some of the issues reported by the build of deprecated Xamarin Android forms PageRenderer.

henrik-me committed:
Build and utest updates

henrik-me committed:
remove bak file

henrik-me committed:
build and restore scripts

henrik-me committed:
adding strongname script

henrik-me committed:
Start of verification script and fix of #834

jennyf19 commented on verify.cmd:
```diff
@@ -0,0 +1,49 @@
+echo Validating available SDK's ...
+if not exist android-25 echo Please install Android SDK v25 (Open SDK Manager from above location)
+if not exist android-26 echo Please install Android SDK v26 (Open SDK Manager from above location)
+if not exist android-27 echo Please install Android SDK v27 (Open SDK Manager from above location)
```
cool...can we install automatically?

henrik-me replied:
Next step will be to kick off the script which will install. For now I'm building out the awareness (and I just added the link to where to install from)

jennyf19 replied:
awesome

jennyf19 commented on strongname.cmd:
```diff
@@ -0,0 +1,21 @@
+
+sn.exe -Vr Microsoft.IdentityModel.Clients.ActiveDirectory.dll
+sn.exe -Vr Microsoft.Identity.Core.dll
+
```
why are you disabling strongname for just this dll?

henrik-me replied:
that would enable the tests to run. Ideally it should be enough to have the dll name and the strongname identifier marked. I will continue testing... and perhaps will identify that I need to do more (so far this seems to work on my box).

jennyf19 replied:
sounds good.

jennyf19 reviewed (changes requested):
Looks good...builds much faster now.
I pulled your branch and did a build w/Core..it was fine, but I did have errors when building ADAL alone w/msbuild (restore/build), but they are not related to the PR (i think)...no issues w/MSAL alone.
I also deployed the AdalAndroidTestApp successfully to a device...but the XForms.Android app returned a null activity and couldn't load. does it work for you w/the changes?

henrik-me commented:
I will take a new look at XForms app. It's likely that the last changes broke something...

henrik-me committed:
Separate shipping bits and corresponding unit tests from build of sample apps. Step 1.

henrik-me committed:
Various build and project updates

henrik-me committed:
Updated build,strongname script added clean script and. Also modified the winrt sample app to ignore fxcop issues. Added a missing ruleset file to one of the projects (needs to be investigated deeper as it seems only to fix debug builds and not release builds).

henrik-me committed:
Merge from dev (Embedded WebView)

henrik-me committed:
various build command updates, merge with dev and winrt project updates

henrik-me committed:
Updating targets for Android testapps and updating build scripts

henrik-me committed:
Merging from the Dev branch

jennyf19 commented:
:shipit: 😸

henrik-me committed:
Adding a change to enable rebuild

henrik-me committed:
Making the AdalAndroidTestApp target v7.1 instead of v8.0 to align with the rest of the test apps. Also added a small comment to the WsTrustRequest class.

henrik-me commented:
Added build scripts for vs dev cmd 2015, vs dev cmd 2017 and std. command-line and also modified various project references and targets. Made a set of additional changes.
